### PR TITLE
Flutter v2 documentation update

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Documentation update for Flutter SDK v2

## Type of documentation
- [ ] New documentation
- [ ] Documentation update
- [ ] Hotfix

## Notion ticket link

> [Flutter v2 Updates](https://www.notion.so/rudderstacks/Documentation-for-Flutter-sdk-V2-069d567720794757b7bce292224c883c)

### CODE REVIEW

**Review by itsdebs — APPROVED** [review]

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx [hunk 1]
```diff
@@ -169,14 +194,68 @@ The `identify` method has the following signature:
...
+<div class="infoBlock">
+
+RudderStack automatically tracks the following optional events:
+<ol>
+<li><code class="inline-code">Application Installed</code></li>
+<li><code class="inline-code">Application Updated</code></li>
+<li><code class="inline-code">Application Opened</code></li>
+<li><code class="inline-code">Application Backgrounded</code></li>
+</ol>
+You can disable these events by calling <code class="inline-code">withTrackLifeCycleEvents(false)</code> on the <code class="inline-code">RudderConfigBuilder</code> object while initializing the RudderStack client. However, we highly recommend keeping them enabled.
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx:247]:
```suggestion
You can disable these events by calling <code class="inline-code">withTrackLifeCycleEvents(false)</code> on the <code class="inline-code">RudderConfigBuilder</code> object while initializing the RudderStack client. However, it is highly recommended to keep them enabled.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx [hunk 2]
```diff
@@ -261,7 +341,12 @@ You can use the `reset` method to clear the persisted `traits` for the `identify
...
 
-## Enabling/Disabling User Tracking via the optOut API \(GDPR Support\)
+## Enabling/disabling user tracking via the `optOut` API(GDPR support)
+
+<div class="infoBlock">
+
+This functionality is not available for the web. For more information, refer to the <a href="#flutter-sdk-api-not-available-for-web">Flutter SDK API not available for the web</a> section below.
+</div>
 
 RudderStack gives the users \(e.g., an EU user\) the ability to opt out of tracking any user activity until the user gives their consent. You can do this by leveraging RudderStack's `optOut` API.
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx:351]:
```suggestion
RudderStack gives the users \(for example, an EU user\) the ability to opt out of tracking any user activity until the user gives their consent. You can do this by leveraging RudderStack's `optOut` API.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx [hunk 3]
```diff
@@ -261,7 +341,12 @@ You can use the `reset` method to clear the persisted `traits` for the `identify
 rudderClient.reset();
 ```
 
-## Enabling/Disabling User Tracking via the optOut API \(GDPR Support\)
+## Enabling/disabling user tracking via the `optOut` API(GDPR support)
```
Comment (ishachopra0401) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx:344]:
```suggestion
## Enabling/disabling the tracking of user activities (GDPR support)
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx [hunk 4]
```diff
@@ -2,127 +2,152 @@
...
 
-- Navigate to your Application's root folder and install all the required dependencies with:
+2. Navigate to your application's root folder and install all the required dependencies with the following command:
 
 ```bash
 flutter pub get
 ```
 
-## Initializing the RudderStack Client
+### Installing the SDK for the web
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply stream-sources/rudderstack-sdk-integration-guides/rudderstack-flutter-sdk/index.mdx:64]: Yep, as discussed these are additional steps to use the SDK for web. Clarified accordingly.

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**